### PR TITLE
[FEATURE] Omnichannel / Directory - Chats Tab

### PR DESCRIPTION
**Pull request by rafaelblink**
## Proposed changes

Adding the chats tab to omnichannel directory.

## Screenshots
image

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by renatobecker** [comment]
Two things about the UI shown in the image attached here that are not following the design system:
- Room tags are missing
- The chat duration column needs to display relative times

**Comment by rafaelblink** [comment]
> Two things about the UI shown in the image attached here that are not following the design system:
> 
> * Room tags are missing
> * The chat duration column needs to display relative times

I've created two tasks to improve this after.

File: client/omnichannel/directory/ChatTab.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		if (sortBy === id) {
+			setSort([id, sortDirection === 'asc' ? 'desc' : 'asc']);
+			return;
+		}
+		setSort([id, 'asc']);
+	});
+
+	const onRowClick = useMutableCallback((_id) => {
+		FlowRouter.go('live', { id: _id });
+		// routing this way causes a 404 that only goes away with a refresh, need to fix in review
```
Comment (renatobecker) [inline client/omnichannel/directory/ChatTab.js:43]: Expecting this fix here..

File: client/omnichannel/directory/ChatTab.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		params={params}
+		renderFilter={({ onChange, ...props }) => <FilterByText onChange={onChange} {...props} />}
+	/>;
+};
+
+
+export default function ChatTab(props) {
+	return <ChatTable {...props} />;
+
+	// return <NotAuthorizedPage />;
```
Comment (renatobecker) [inline client/omnichannel/directory/ChatTab.js:91]: ???

File: client/omnichannel/directory/ChatTab.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		setSort([id, 'asc']);
+	});
+
+	const onRowClick = useMutableCallback((_id) => {
+		FlowRouter.go('live', { id: _id });
+		// routing this way causes a 404 that only goes away with a refresh, need to fix in review
+		// livechatRoomRoute.push({ id: _id });
+	});
+
+	const { data } = useEndpointDataExperimental('livechat/rooms', query) || {};
```
Comment (renatobecker) [inline client/omnichannel/directory/ChatTab.js:47]: This call is returning nothing when the user is just a `livechat agent`, I mean, when the user is a `livechat manager` the endpoint is working properly.

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: client/omnichannel/directory/ContactTab.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+	const debouncedParams = useDebouncedValue(params, 500);
+	const debouncedSort = useDebouncedValue(sort, 500);
+	const query = useQuery(debouncedParams, debouncedSort);
+
+	const onHeaderClick = useMutableCallback((id) => {
+		const [sortBy, sortDirection] = sort;
+
+		if (sortBy === id) {
+			setSort([id, sortDirection === 'asc' ? 'desc' : 'asc']);
```
[the remark this answers is not in the document]
Comment (rafaelblink) [reply client/omnichannel/directory/ContactTab.js:30]: Is in Contact branch.

File: client/omnichannel/directory/ContactTab.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		setParams={setParams}
+		params={params}
+		renderFilter={({ onChange, ...props }) => <FilterByText onChange={onChange} {...props} />}
+	/>;
+}
+
+export default function ContactTab(props) {
+	return <ContactTable {...props} />;
+
+	// return <NotAuthorizedPage />;
```
[the remark this answers is not in the document]
Comment (rafaelblink) [reply client/omnichannel/directory/ContactTab.js:67]: It's on Contact branch.

File: client/omnichannel/directory/ChatTab.js (hunk 1 above)
Comment (rafaelblink) [reply client/omnichannel/directory/ChatTab.js]: Comment was brought from another page, sorry, this will open the context bar in the next sprints.

**Pull request closed without merging**